### PR TITLE
Convert some internal engine classes to records

### PR DESCRIPTION
**Pull request by thecoop**
Small refactoring to convert some POJO internal engine classes to records

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/translog/Translog.java [hunk 1]
```diff
@@ -908,59 +908,21 @@ public TranslogDeletionPolicy getDeletionPolicy() {
...
             }
-            if (translogLocation != location.translogLocation) {
-                return false;
-            }
-            return size == location.size;
-
-        }
-
-        @Override
-        public int hashCode() {
```
Comment (kingherc) [inline server/src/main/java/org/elasticsearch/index/translog/Translog.java:960]: We never store this, right? And thus we can live with whatever implementation the JVM provides for records hashCode. Or should we re-use the same hashCode to be safe?

**Review by kingherc — APPROVED** [review]
LGTM. Just one comment to review to ensure we don't have a pitfall.

File: server/src/main/java/org/elasticsearch/index/translog/Translog.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/index/translog/Translog.java:1016]: The tests pass with `hashCode` poisoned, so it looks like it isn't used at all

**Pull request merged**